### DISCUSSION TITLE
Make the MTD attach to the best router on reset

### DISCUSSION THREAD

**Discussion by fgervais** [openthread/openthread#9084] [Q&A] [answered]
I have MTDs deployed in an environment with multiple BRs+TREL.

When a MTD starts for the first time it will bind to what is the best router at the time.

However a new border router might come up which might be better for some MTDs. These MTDs don't run `Periodic Parent Search` and so won't try to bind to the new router. This is fine.

However these MTDs have a watchdog which will reset the chip if some high level condition is not met. For exemple if it cannot successfully ping the MQTT server.

On reset I though these devices would bind to the best router but it doesn't seem like they do. They just seem to bind to the previous parent. I assume that this is seen a a feature since they can continue where the left off. Also presumably finish the attach hand-shaking faster.

However in my case I would like them to pick a fresh best parent on reset. Any suggestions on how I could achieve this?

**Comment by jwhui** [comment, marked as answer]
> On reset I though these devices would bind to the best router but it doesn't seem like they do. They just seem to bind to the previous parent. I assume that this is seen a a feature since they can continue where the left off. Also presumably finish the attach hand-shaking faster.

Correct. [Thread 1.3.0](https://www.threadgroup.org/support#specifications) Section 4.6.4 describes the reattach process. The specification currently requires this behavior, so any Thread certified devices must demonstrate it as part of testing.

> However in my case I would like them to pick a fresh best parent on reset. Any suggestions on how I could achieve this?

The parent information is stored in non-volatile memory:
https://github.com/openthread/openthread/blob/afe5041d075647a2e6c812c04cbd803220b3b215/include/openthread/platform/settings.h#L67

You could try deleting the parent info setting. Note, however, that doing so would not be conformant with the Thread 1.3.0 specification.

If you would like to advocate for updating the Thread Specification, I would encourage you to submit your idea to the Thread Group.

**Reply by abtink** [reply]
Somewhat related to this, you can also consider enabling and using [`OPENTHREAD_CONFIG_PARENT_SEARCH_ENABLE` feature](https://openthread.io/guides/build/features/periodic-parent-search).
